app/internal/actor/attorney.go

```diff
@@ -28,50 +28,112 @@ func (a Attorney) FullName() string {
 	return fmt.Sprintf("%s %s", a.FirstNames, a.LastName)
 }
 
-type Attorneys []Attorney
+// TrustCorporation contains details about a trust corporation, provided by the applicant
+type TrustCorporation struct {
+	// Name of the company
+	Name string
+	// CompanyNumber as registered by Companies House
+	CompanyNumber string
+	// Email to contact the company
+	Email string
+	// Address of the company
+	Address place.Address
+}
+
+type Attorneys struct {
+	TrustCorporation TrustCorporation
+	Attorneys        []Attorney
+}
+
+func (as Attorneys) Len() int {
+	if as.TrustCorporation.Name == "" {
+		return len(as.Attorneys)
+	}
+
+	return len(as.Attorneys) + 1
+}
+
+func (as Attorneys) Complete() bool {
+	if as.TrustCorporation.Name != "" && as.TrustCorporation.Address.Line1 == "" {
+		return false
+	}
+
+	for _, a := range as.Attorneys {
+		if a.FirstNames == "" || (a.Address.Line1 == "" && a.Email == "") {
+			return false
+		}
+	}
+
+	return true
+}
+
+func (as Attorneys) Addresses() []place.Address {
+	var addresses []place.Address
+
+	if as.TrustCorporation.Address.String() != "" && !slices.Contains(addresses, as.TrustCorporation.Address) {
```

Comment:
Won't `addresses` always be an empty slice here?

Reply:
true true